src/common/snippets/src/op/serialization_node.cpp

```diff
@@ -40,22 +40,58 @@ std::shared_ptr<Node> SerializationNode::clone_with_new_inputs(const OutputVecto
 }
 
 bool SerializationNode::visit_attributes(AttributeVisitor &visitor) {
+    auto is_planar_layout = [](const std::vector<size_t>& layout) {
+        for (size_t i = 0; i < layout.size(); ++i)
+            if (layout[i] != i) return false;
+        return true;
+    };
+    auto subtensor2str = [](const VectorDims& subtensor) {
+        std::stringstream ss;
+        for (size_t i = 0; i < subtensor.size(); ++i) {
+            const auto& v = subtensor[i];
+            const auto v_str = (v == lowered::PortDescriptor::ServiceDimensions::FULL_DIM) ? "FULL_DIM" : std::to_string(v);
+            const auto del = i < subtensor.size() - 1 ? ", " : "";
+            ss << v_str << del;
+        }
+        return ss.str();
+    };
+
     std::vector<size_t> in_regs, out_regs;
     std::vector<std::string> in_reg_types, out_reg_types;
-    std::vector<std::pair<std::string, std::vector<size_t>>> shapes;
+    std::vector<std::pair<std::string, ov::PartialShape>> shapes;
+    std::vector<std::pair<std::string, std::string>> subtensors;
+    std::vector<std::pair<std::string, std::vector<size_t>>> layouts;
     for (size_t i = 0; i < m_expr->get_input_count(); i++) {
         const auto& desc = m_expr->get_input_port_descriptor(i);
-        const auto &shape = desc->get_shape();
+        const auto& shape = desc->get_shape();
         if (!shape.empty())
-            shapes.emplace_back("in_shape_" + std::to_string(i), shape);
+            shapes.emplace_back("in_shape_" + std::to_string(i), ov::PartialShape(shape));
```

Comment:
It's better to use `vdims_to_pshape` from utils: only in this case we will get "?" instead of a huge value. Sorry, I should have mentioned that right away.

Reply:
I verified this moment and the current solution works. So I think we can just call ctor of `PartialShape` as it's done 😃

Reply:
Wow, that's interesting, thanks for the finding. Then I don't understand why do we need the `vdims_to_pshape` helper 😄